### ISSUE TITLE
ConnectionManager not sending messages to server

### ISSUE THREAD

**Issue by snaulX** [Facepunch/Facepunch.Steamworks#511]
I work in Unity. I have ConnectionManager `game` in one script and SocketManager `server` in other script. In these scripts in Update methods I call `game.Receive();` and `server.Receive();` methods. And when server send messages - all works but when client send messages - these messages dont reach the server.

Some code from client script:
```cs
void Start()
    {
        game = SteamNetworkingSockets.ConnectNormal<GameClient>(NetAddress.LocalHost(3000)); // GameClient is ConnenctionManager
        Debug.Log($"Client: {game}");
    }

    private void Update()
    {
        game.Receive();
    }

public void Fire()
{
        Result result = game.Connection.SendMessage(new byte[] { 0 });
        Debug.Log($"Fire was sended to server: {result}"); // this always OK
}
```

Some code from server script:
```cs
private void Start()
{
	server = SteamNetworkingSockets.CreateNormalSocket<GameServer>(NetAddress.LocalHost(3000));
	Debug.Log($"Server: {server}");
}

private void Update()
{
	server.Receive();
}

private void OnDestroy()
{
	try
	{
		server.Close();
	}
	catch (NullReferenceException)
        {
		//UnityEngine.Debug.LogError("Server was null");
        }
}

private void SendCommand(IntPtr command, int size)
{
	foreach (Connection c in server.Connected)
	{
		Result r = c.SendMessage(command, size);
		Debug.Log($"Command was sended to client with id {c.Id} with result {r}");
	}
}
```
What I doing wrong?

**Comment by bthomas2622** [comment]
having the exact same issue, were you ever able to solve?

**Comment by snaulX** [comment]
When it solves I will write solution here and close the issue. But I don't solved this issue and really don't understand how to fix it. So if you have solution - please write here.

**Comment by bthomas2622** [comment]
@snaulX got a fix! See #467

**Comment by snaulX** [comment]
I have other issue but it don't linked with this. So I close this issue.
